spec/project-spec.coffee

```diff
@@ -436,6 +436,29 @@ describe "Project", ->
       expect(atom.project.getPaths()).toEqual([path.join(__dirname, "..", "src")])
       expect(atom.project.getRepositories()[0].isSubmodule("src")).toBe false
 
+    it "removes a path that is represented as a URI", ->
+      ftpUri = "ftp://example.com/some/folder"
+      directoryProvider =
+        directoryForURISync: (uri) ->
+          if uri is ftpUri
```

Comment:
Since this `directoryProvider` is just here to allow us to set `ftpURI` as a path, I would get rid of the `if`/`else`/`throw` logic in `directoryForURISync` and simply return your fake `Directory` unconditionally. That way the test can get straight to the point.